v2/datastream-to-spanner/src/test/java/com/google/cloud/teleport/v2/templates/failureinjectiontesting/DataStreamToSpannerCDCFT.java

```diff
@@ -0,0 +1,227 @@
+      throws IOException, InterruptedException, SQLException, ExecutionException {
+    String testRootDir = getClass().getSimpleName();
+    // create subscriptions
+    String gcsPrefix =
+        String.join("/", new String[] {testRootDir, gcsResourceManager.runId(), testName, "cdc"});
+    SubscriptionName subscription =
+        createPubsubResources(
+            testRootDir + testName, pubsubResourceManager, gcsPrefix, gcsResourceManager);
+
+    String dlqGcsPrefix =
+        String.join("/", new String[] {testRootDir, gcsResourceManager.runId(), testName, "dlq"});
+    SubscriptionName dlqSubscription =
+        createPubsubResources(
+            testRootDir + testName + "dlq",
+            pubsubResourceManager,
+            dlqGcsPrefix,
+            gcsResourceManager);
+    String artifactBucket = TestProperties.artifactBucket();
+
+    // launch datastream
+    DatastreamResourceManager.Builder datastreamResourceManagerBuilder =
+        DatastreamResourceManager.builder(testName, PROJECT, REGION)
+            .setCredentialsProvider(credentialsProvider);
+    if (System.getProperty("privateConnectivity") != null) {
+      datastreamResourceManagerBuilder.setPrivateConnectivity(
+          System.getProperty("privateConnectivity"));
+    }
+    datastreamResourceManager = datastreamResourceManagerBuilder.build();
+    Stream stream =
+        createDataStreamResources(
+            artifactBucket, gcsPrefix, sourceConnectionProfile, datastreamResourceManager);
+
+    int numRows = 100;
+    int burstIterations = 10000;
+
+    // generate Load
+    CDCCorrectnessTestUtil testUtil = new CDCCorrectnessTestUtil();
+    testUtil.generateLoad(numRows, burstIterations, sourceDBResourceManager);
+
+    FlexTemplateDataflowJobResourceManager.Builder flexTemplateBuilder =
+        FlexTemplateDataflowJobResourceManager.builder(testName)
+            .addEnvironmentVariable("numWorkers", 10)
+            .addEnvironmentVariable("maxWorkers", 20);
+
+    // launch dataflow template
+    PipelineLauncher.LaunchInfo jobInfo =
+        launchFwdDataflowJob(
+            spannerResourceManager,
+            gcsPrefix,
+            stream.getName(),
+            dlqGcsPrefix,
+            subscription.toString(),
+            dlqSubscription.toString(),
+            flexTemplateBuilder,
+            shadowTableSpannerResourceManager);
+
+    // Wait for Forward migration job to be in running state
+    assertThatPipeline(jobInfo).isRunning();
+
+    // Wait for at least 1 row to appear in Spanner
+    ConditionCheck conditionCheck =
+        SpannerRowsCheck.builder(spannerResourceManager, USERS_TABLE).setMinRows(1).build();
+    PipelineOperator.Result result =
+        pipelineOperator()
+            .waitForCondition(createConfig(jobInfo, Duration.ofMinutes(20)), conditionCheck);
+    assertThatResult(result).meetsConditions();
+
+    DataflowFailureInjector.abruptlyKillWorkers(jobInfo.projectId(), jobInfo.jobId());
```

Comment:
this should happen a few times during the duration of the test. Probably after the job starts. Considering the job will start after about 3 minutes, this should be called after a short while. It would be better if this checks running status before killing the workers

Reply:
Done. Modified the code to kill workers 4 times at the interval of 20 seconds each. This actually happens after the job starts. The conditional wait at Line 198 to 203 is for waiting till 1 row appears in spanner indicating the pipeline has fully started.